src/components/permissions/HasPermission.js

```diff
@@ -0,0 +1,38 @@
+"use client";
+import React from "react";
+import { useGetRolesQuery } from "@/redux/services/roleApi";
+import Spinner from '@/components/Spinner';
+import ServerErrorComponent from "@/components/ServerError";
+import { usePathname } from "next/navigation";
+import { methods } from "@/app/constants/constants";
+
+const HasPermission = ({children, to:permissionTo=methods.read}) => {
+    const { data: rolesData, isLoading, isError } = useGetRolesQuery();
+    const pathName = usePathname();
+    const token = localStorage.getItem('token');
+    if(token && rolesData) {
+      const claims = JSON.parse(atob(token.split('.')[1]));
+      const pagePermissions = rolesData.filter(({roleName}) => claims.roles.find(currentRole => currentRole === roleName)).flatMap(({permissions}) => permissions);
```

Comment:
this line is too long maybe you could consider to separate like this:
`const pagePermissions = rolesData.filter(({roleName}) => 
  claims.roles.find(currentRole =>
  currentRole === roleName)).flatMap(({permissions}) => permissions);`

Reply:
ok